### PR TITLE
[Merge with 0.3.4] Update CRD apiversion to v1

### LINKED ISSUE #25: Update CRD to 'apiextensions.k8s.io/v1' and add schema
`apiextensions.k8s.io/v1beta1` is unavailable starting from Kubernetes 1.22+. We need to update it to 'apiextensions.k8s.io/v1'. This requires also adding `schema` for our resource. I guess it should be possible to generate it from the Helm Chart.

### PR DESCRIPTION
**Pull request by hasancelik**
fixes #25 

@mtyazici, you need to add  `--crd-version=v1` flag fot `operator-sdk new` command at release actions:
`operator-sdk new  OPERATOR_NAME --api-version=hazelcast.com/v1alpha1 --crd-version=v1 ...`

### CODE REVIEW

**Review by leszko — APPROVED** [review]
Thanks @hasancelik

File: hazelcast-enterprise-operator/hazelcastcluster.crd.yaml [hunk 1]
```diff
@@ -1,11 +1,7 @@
-apiVersion: apiextensions.k8s.io/v1beta1
+apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   name: hazelcastenterprises.hazelcast.com
-  labels:
```
Comment (mtyazici) [inline hazelcast-enterprise-operator/hazelcastcluster.crd.yaml:5]: Aren't these things necessary for testing? In the workflows I use label information.

Comment (hasancelik) [reply hazelcast-enterprise-operator/hazelcastcluster.crd.yaml:5]: Afaik, release script just checking labels of pods not CRDs. CR/Pod have those labels already btw --> https://github.com/hazelcast/hazelcast-operator/blob/master/hazelcast-operator/hazelcast.yaml#L5-L8

Comment (mtyazici) [reply hazelcast-enterprise-operator/hazelcastcluster.crd.yaml:5]: Now I took a look and you are right, I will delete these labels from `crd`s in my PR then.

**Review by mtyazici — APPROVED** [review]

**Pull request merged**